### DISCUSSION TITLE
I can't get t macro working in Next.js with Typescript

### DISCUSSION THREAD

**Discussion by crazyyi** [lingui/js-lingui#1521] [Q&A]
I ended up switching to `i18n._` function instead. 
What is t macro for anyway?
I found that `i18n._` is doing exactly the same thing, and it works in Next.js with Typescript.

**EDIT:**
Just found out using `i18n._` wouldn't extract language keys into `po` file. When doing `lingui extract` it would give me this error if I don't use `t` macro:
```
Missing message ID, skipping.
```

**Comment by timofei-iatsenko** [comment]
read https://lingui.dev/tutorials/react#macros-vs-components
check examples https://lingui.dev/ref/macro#t 

Everything work perfectly with nextjs and typescript and any other technology. It battle tested in production for years. If it doesn't work in your case - look for the issue in your app code.

Probably you're trying to put some dynamic expression or variable instead of a message. Hard to say without looking at your code.

**Reply by crazyyi** [reply]
@thekip 
It's not about the macro itself. It's about how to config that to make it work.
I am following this [example](https://github.com/lingui/swc-plugin) and config exactly like that. But unfortunately, that example doesn't use the t macro.
So I downloaded that [example](https://github.com/lingui/swc-plugin), added t macro examples in `index.tsx`, and did a `yarn build`, then ran `yarn start`, 
I have forked your example and uploaded the [new example](https://github.com/crazyyi/swc-plugin-test) to my repo with added t macro examples (in yellow color) which are similiar to what I use in my current project. Some of the t macros don't get translated in production mode (yarn start) Can you have a look at the issue please?

**Reply by timofei-iatsenko** [reply]
1. if your just seting up new project - use v4 right now, don't need to wait for official release.  
2. Your code is wrong. How do you thing strings outside of function being assigned to constants should magically change it's own values? It's just a javascript, not releated to macro, t, lingui whatever. Read here https://lingui.dev/tutorials/react-patterns#lazy-translations

**Reply by crazyyi** [reply]
@thekip 
 - How to use v4 without the publicly available package? Should I just download the zip files and put them somewhere in my project?
 - I am not sure I understand what you say here. Why does `<Trans>` work outside of function but t doesn't work? What is the key difference between `<Trans>` and t? I thought t is used for translating strings because it returns a string.

**Reply by timofei-iatsenko** [reply]
```json
{
 "dependecies": {
    "@lingui/core": "^4.0.0-next.1",
  }
}
```

> I am not sure I understand what you say here. Why does <Trans> work outside of function but t doesn't work? What is the key difference between <Trans> and t? I thought t is used for translating strings because it returns a string.

Please, learn Javascript. I'm not here to teach people basics. If you don't understand - just copy from docs - that would work.
